Thu Jan  4 14:49:11 CET 2007  Dafydd Harries <dafydd.harries@collabora.co.uk>
  * presence cache: update cache *before* trying to process nickname/avatar/caps info, as these functions need the presence object
diff -rN -u old-telepathy-gabble/src/gabble-presence-cache.c new-telepathy-gabble/src/gabble-presence-cache.c
--- old-telepathy-gabble/src/gabble-presence-cache.c	2007-01-25 15:09:31.000000000 +0100
+++ new-telepathy-gabble/src/gabble-presence-cache.c	2007-01-25 15:09:35.000000000 +0100
@@ -987,14 +987,14 @@
     {
     case LM_MESSAGE_SUB_TYPE_NOT_SET:
     case LM_MESSAGE_SUB_TYPE_AVAILABLE:
-      _grab_nickname (cache, handle, from, presence_node);
-      _grab_avatar_sha1 (cache, handle, from, presence_node);
-      _process_caps (cache, handle, from, presence_node);
-
       presence_id = _presence_node_get_status (presence_node);
       gabble_presence_cache_update (cache, handle, resource, presence_id,
           status_message, priority);
 
+      _grab_nickname (cache, handle, from, presence_node);
+      _grab_avatar_sha1 (cache, handle, from, presence_node);
+      _process_caps (cache, handle, from, presence_node);
+
       ret = LM_HANDLER_RESULT_REMOVE_MESSAGE;
       break;
 

